pkg/proxy/downstream.go

```diff
@@ -749,7 +760,7 @@ func (s *downStream) setupRetry(endStream bool) bool {
 
 // Note: retry-timer MUST be stopped before active stream got recycled, otherwise resetting stream's properties will cause panic here
 func (s *downStream) doRetry() {
-	pool, err := s.initializeUpstreamConnectionPool(s.cluster.Name(), nil)
+	pool, err := s.initializeUpstreamConnectionPool(s)
```

Comment:
Should putClusterSnapshot and Get again？

Reply:
Get is called in doReceiveHeaders